packages/hardware-trezor/src/TrezorKeyAgent.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as Crypto from '@cardano-sdk/crypto';
+import * as Trezor from 'trezor-connect';
+import { Cardano, coreToCml } from '@cardano-sdk/core';
+import { CreateTrezorKeyAgentProps } from '@cardano-sdk/key-management/dist/cjs/TrezorKeyAgent';
+import { TrezorKeyAgent as DeprecatedTrezorKeyAgent, KeyAgentDependencies, errors } from '@cardano-sdk/key-management';
+import { ManagedFreeableScope } from '@cardano-sdk/util';
+import { txToTrezor } from './transformers/tx';
+import TrezorConnect from 'trezor-connect';
+
+const transportTypedError = (error?: any) =>
+  new errors.AuthenticationError(
+    'Trezor transport failed',
+    new errors.TransportError('Trezor transport failed', error)
+  );
+
+export class TrezorKeyAgent extends DeprecatedTrezorKeyAgent {
```

Comment:
@DominikGuzei @tomislavhoracek just to check, are we going to remove this `extends` part in [LW-6803](https://input-output.atlassian.net/browse/LW-6803)?